custom/icds/forms.py

```diff
@@ -175,3 +193,49 @@ def submit(self, domain, email):
                             self.cleaned_data['month'],
                             self.cleaned_data['location_id'],
                             email)
+
+
+class CustomSMSReportRequestForm(forms.Form):
+
+    date_range = forms.CharField(
+        label="Select Date Range",
+        widget=forms.TextInput(
+            attrs={'class': 'form-control', 'id': 'date_range_selector'}
+        ),
+        required=True
+    )
+
+    start_date = forms.CharField(widget=forms.HiddenInput(
+        attrs={'id': 'report_start_date'}
+    ), required=True)
+    end_date = forms.CharField(widget=forms.HiddenInput(
+        attrs={'id': 'report_end_date'}
+    ), required=True)
+
+    def __init__(self, *args, **kwargs):
+        super(CustomSMSReportRequestForm, self).__init__(*args, **kwargs)
+        self.helper = FormHelper()
+        self.helper.form_method = 'post'
+        self.helper.layout = crispy.Layout(
+            hqcrispy.Field('date_range'),
+            hqcrispy.Field('start_date'),
+            hqcrispy.Field('end_date'),
+            twbscrispy.StrictButton(
+                _('Generate Report'),
+                type='submit',
+                css_class='btn-primary',
+            )
+        )
+
+    def clean_start_date(self):
+        data = self.cleaned_data['start_date']
+        validate_date(data)
+        return data
+
+    def clean_end_date(self):
+        start_date = self.cleaned_data['start_date']
+        end_date = self.cleaned_data['end_date']
+        validate_date(end_date)
+        if(start_date < end_date):
```

Comment:
Actually the current check will not allow one-day reports, I was not sure whether we will be using one-day reports or not. 
I will update the condition, I don't think this will do any harm 😄